### PR TITLE
Test framework user helpers

### PR DESCRIPTION
**Pull request by igomez06**

### CODE REVIEW

**Review by paynejacob — CHANGES_REQUESTED** [review]
Can you make this pr just the user framework changes.  This includes a lot more than the user helpers.

**Comment by igomez06** [comment]
> Can you make this pr just the user framework changes. This includes a lot more than the user helpers.

Once we get the other PR merged it will clean up this one. I branched off our the wait/watch branch.

File: tests/framework/extensions/users/users.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+}
+
+func AddProjectMember(rancherClient *rancher.Client, project *management.Project, user *management.User, projectRole string) (*management.ProjectRoleTemplateBinding, error) {
+	role := &management.ProjectRoleTemplateBinding{
+		ProjectID:       project.ID,
+		UserPrincipalID: user.PrincipalIDs[0],
+		RoleTemplateID:  projectRole,
+	}
+
+	return rancherClient.Management.ProjectRoleTemplateBinding.Create(role)
```
Comment (paynejacob) [inline tests/framework/extensions/users/users.go:31]: add a wait to ensure the project namespace exists

File: tests/framework/extensions/users/users.go [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+}
+
+func RemoveProjectMember(rancherClient *rancher.Client, role *management.ProjectRoleTemplateBinding) error {
```
Comment (paynejacob) [inline tests/framework/extensions/users/users.go:34]: accept user in args here not bidng.

File: tests/framework/extensions/users/users.go [hunk 3]
```diff
@@ -0,0 +1,36 @@
+package users
+
+import (
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	management "github.com/rancher/rancher/tests/framework/clients/rancher/generated/management/v3"
+)
+
+func CreateUserWithRole(rancherClient *rancher.Client, user *management.User, role *management.GlobalRoleBinding) (*management.User, error) {
```
Comment (paynejacob) [inline tests/framework/extensions/users/users.go:8]: accept role and create the role binding for the user

**Review by paynejacob — CHANGES_REQUESTED** [review]

**Pull request closed without merging**